### PR TITLE
Update for Facebook on Android version 7.1.0

### PR DESCRIPTION
**Pull request by bholmes**

### CODE REVIEW

File: Facebook.Android/source/facebook-core/additions/additions.cs [hunk 1]
```diff
@@ -1,12 +1,65 @@
...
+			IntPtr __ret = JNIEnv.ToLocalJniHandle(__this.DoInBackground(p0));
+			if (p0 != null)
+				JNIEnv.CopyArray(p0, native_p0);
+			return __ret;
+		}
+		#pragma warning restore 0169
+
+		// Metadata.xml XPath method reference: path="/api/package[@name='com.facebook.appevents.internal']/class[@name='FileDownloadTask']/method[@name='doInBackground' and count(parameter)=1 and parameter[1][@type='java.lang.String...']]"
+		[Register("doInBackground", "([Ljava/lang/String;)Ljava/lang/Boolean;", "GetDoInBackground_arrayLjava_lang_String_Handler")]
+		protected override unsafe global::Java.Lang.Object DoInBackground(params global::Java.Lang.Object[] p0)
```
Comment (mattleibow) [inline Facebook.Android/source/facebook-core/additions/additions.cs:44]: What was the reason for removing this and then adding it back manually? Is there some compiler error?

Comment (bholmes) [reply Facebook.Android/source/facebook-core/additions/additions.cs:44]: I could not get the generator to use the override keyword.  All other things did line up AFAICT

Comment (mattleibow) [reply Facebook.Android/source/facebook-core/additions/additions.cs:44]: Awesome. CI will tell us if anything is bad. I expect only good things from CI... _slaps roof of devops_

**Comment by mattleibow** [comment]
Something went wrong with the parameter names...

# API diff: Xamarin.Facebook.Share.Android.dll

## Xamarin.Facebook.Share.Android.dll

### Namespace Xamarin.Facebook.Internal

#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper

#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper.ErrorEventArgs

Modified constructors:

```diff
-public CollectionMapper.ErrorEventArgs (Xamarin.Facebook.FacebookException exception)
+public CollectionMapper.ErrorEventArgs (Xamarin.Facebook.FacebookException p0)
```

Removed property:

```csharp
public Xamarin.Facebook.FacebookException Exception { get; }
```


#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper.ICollection

Modified methods:

```diff
-public abstract Java.Lang.Object Get (Java.Lang.Object key)
+public abstract Java.Lang.Object Get (Java.Lang.Object p0)
-public abstract void Set (Java.Lang.Object key, Java.Lang.Object value---p1---, CollectionMapper.IOnErrorListener onErrorListener---p2---)
+public abstract void Set (Java.Lang.Object p0, Java.Lang.Object +++value+++p1, CollectionMapper.IOnErrorListener +++onErrorListener+++p2)
```


#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper.IOnErrorListener

Modified methods:

```diff
-public abstract void OnError (Xamarin.Facebook.FacebookException exception)
+public abstract void OnError (Xamarin.Facebook.FacebookException p0)
```


#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper.IOnMapValueCompleteListener

Modified methods:

```diff
-public abstract void OnComplete (Java.Lang.Object mappedValue)
+public abstract void OnComplete (Java.Lang.Object p0)
```


#### Type Changed: Xamarin.Facebook.Internal.CollectionMapper.IValueMapper

Modified methods:

```diff
-public abstract void MapValue (Java.Lang.Object value, CollectionMapper.IOnMapValueCompleteListener onMapValueCompleteListener---p1---)
+public abstract void MapValue (Java.Lang.Object p0, CollectionMapper.IOnMapValueCompleteListener +++onMapValueCompleteListener+++p1)
```




### Namespace Xamarin.Facebook.Share

#### Type Changed: Xamarin.Facebook.Share.DeviceShareDialog

Removed constructor:

```csharp
public DeviceShareDialog (Android.Support.V4.App.Fragment fragment);
```


#### Type Changed: Xamarin.Facebook.Share.ShareApi

Modified methods:

```diff
-public void Share (Xamarin.Facebook.IFacebookCallback callback)
+public void Share (Xamarin.Facebook.IFacebookCallback p0)
-public void Share (Model.ShareContent shareContent, Xamarin.Facebook.IFacebookCallback callback---p1---)
+public void Share (Model.ShareContent p0, Xamarin.Facebook.IFacebookCallback +++callback+++p1)
```



### Namespace Xamarin.Facebook.Share.Internal

#### Type Changed: Xamarin.Facebook.Share.Internal.VideoUploader

Removed method:

```csharp
public static void UploadAsync (Xamarin.Facebook.Share.Model.ShareVideoContent videoContent, Xamarin.Facebook.IFacebookCallback callback);
```



### Namespace Xamarin.Facebook.Share.Widget

#### Type Changed: Xamarin.Facebook.Share.Widget.AppInviteDialog

Removed method:

```csharp
[Obsolete ("deprecated")]
public static void Show (Android.Support.V4.App.Fragment fragment, Xamarin.Facebook.Share.Model.AppInviteContent appInviteContent);
```


#### Type Changed: Xamarin.Facebook.Share.Widget.CreateAppGroupDialog

Removed method:

```csharp
[Obsolete ("deprecated")]
public static void Show (Android.Support.V4.App.Fragment fragment, Xamarin.Facebook.Share.Model.AppGroupCreationContent appGroupCreationContent);
```


#### Type Changed: Xamarin.Facebook.Share.Widget.GameRequestDialog

Removed constructor:

```csharp
public GameRequestDialog (Android.Support.V4.App.Fragment fragment);
```

Removed method:

```csharp
public static void Show (Android.Support.V4.App.Fragment fragment, Xamarin.Facebook.Share.Model.GameRequestContent gameRequestContent);
```


#### Type Changed: Xamarin.Facebook.Share.Widget.JoinAppGroupDialog

Removed method:

```csharp
[Obsolete ("deprecated")]
public static void Show (Android.Support.V4.App.Fragment fragment, string groupId);
```


#### Type Changed: Xamarin.Facebook.Share.Widget.MessageDialog

Removed constructor:

```csharp
public MessageDialog (Android.Support.V4.App.Fragment fragment);
```

Removed method:

```csharp
public static void Show (Android.Support.V4.App.Fragment fragment, Xamarin.Facebook.Share.Model.ShareContent shareContent);
```

**Comment by bholmes** [comment]
Facebook must be using a javadoc format that we can not parse.  Need to get someone to look at fixing the parser or getting the parameter names another way.

**Comment by bholmes** [comment]
> g a javadoc format that we can not parse. Need to get someone to look at fixing the parser or getting the parameter names another way. Adding do-not-merge

Fixed by adding a custom java doc parser to this repo.  The tool is run manually and results committed to source.

**Pull request merged**